agent/worker.go

```diff
@@ -278,8 +278,14 @@ func reconcileTaskState(ctx context.Context, w *worker, assignments []*api.Assig
 
 	removeTaskAssignment := func(taskID string) error {
 		ctx := log.WithLogger(ctx, log.G(ctx).WithField("task.id", taskID))
-		if err := SetTaskAssignment(tx, taskID, false); err != nil {
-			log.G(ctx).WithError(err).Error("error setting task assignment in database")
+		// if a task is no longer assigned, then we do not have to keep track
+		// of it. a task will only be unassigned when it is deleted on the
+		// manager. insteaad of SetTaskAssginment to true, we'll just remove
+		// the task now.
+		if err := DeleteTask(tx, taskID); err != nil {
+			log.G(ctx).WithError(err).Errorf(
+				"error removing de-assigned task %v", taskID,
+			)
 		}
 		return err
```

Comment:
what error is returned here? if this is supposed to be an error that was set somewhere above, we should have a different name for this variable

otherwise, this function should probably not return an error at all

Reply:
🤔  yeah that's not right, fixing that.

Reply:
@dperny this one was not yet addressed, correct?

Reply:
Yeah, it's fixed, I just did so in a very awkward way. Let me make it the more typical pattern though.

Reply:
the scope of `err` is now correct here, but i was so laser focused on that problem that i overlooked the more familiar

```go
if err != nil {
    return err
}
return nil
```
pattern.

Reply:
oh! you're right; it's just github that showed it weird. 👍